lib/OpenQA/Utils.pm

```diff
@@ -573,5 +574,31 @@ sub wait_with_progress {
     print "\n";
 }
 
+sub mark_job_linked {
+    my ($jobid, $referer_url) = @_;
+
+    my $referer = Mojo::URL->new($referer_url)->host;
+    if ($referer && grep { $referer eq $_ } @{$app->config->{global}->{recognized_referers}}) {
```

Comment:
I would like to see some log_debug in case there is a referere but not recognized. To give the admin a chance to fix his config

The rest looks good to me